examples/gno.land/r/demo/users/z_11_filetest.gno

```diff
@@ -0,0 +1,38 @@
+package main
+
+// SEND: 2000000000ugnot
+
+import (
+	"strconv"
+
+	"gno.land/r/demo/users"
+)
+
+func main() {
+	users.Register("", "gnouser", "my profile")
+
+	{
+		// Normal usage
+		names := users.ListUsersByPrefix("a", 10)
+		println("# names: " + strconv.Itoa(len(names)))
+		println("name: " + names[0])
+	}
+
+	{
+		// The prefix is after "gnouser"
+		names := users.ListUsersByPrefix("go", 10)
+		println("# names: " + strconv.Itoa(len(names)))
+	}
+
+	{
+		// Reach maxResults
+		names := users.ListUsersByPrefix("a", 0)
+		println("# names: " + strconv.Itoa(len(names)))
+	}
+}
+
+// Output:
+// # names: 1
+// name: gnouser
```

Comment:
why is `gnouser` found for `prefix="a"`?

Reply:
The latest commit limits results to the prefix.